src/popup-page/modules/postage-batch-element.tsx

```diff
@@ -0,0 +1,95 @@
+import { PostageBatch } from '@ethersphere/bee-js'
+import React, { ReactElement, useContext, useEffect, useState } from 'react'
+import { getPostageBatches } from '../../utils/bee-js'
+import { GlobalContext } from '../context/global'
+
+export function PostageBatchElement(): ReactElement {
+  const globalStateContext = useContext(GlobalContext)
+  const { dispatch: dispatchGlobalState, state: globalState } = globalStateContext
+  const [fetchedPostageBatches, setFetchedPostageBatches] = useState<PostageBatch[]>([])
+
+  const init = (): void => {
+    console.log('globalstate', globalState.globalPostageBatchEnabled)
+
+    if (globalState.globalPostageBatchEnabled) retrievePostageBatches()
+  }
+
+  useEffect(() => {
+    init()
+  }, [globalState.globalPostageBatchEnabled])
+
+  const handleUseGlobalPostageBatch = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const { checked } = event.target
+    dispatchGlobalState({ type: 'GLOBAL_POSTAGE_BATCH_ENABLED_SAVE', newValue: checked })
+
+    if (checked) {
+      retrievePostageBatches()
+    }
+  }
+
+  const retrievePostageBatches = async () => {
+    console.log('fetch postagethings')
+    setFetchedPostageBatches(await getPostageBatches(globalState.beeApiUrl))
+  }
+
+  const postageBatchElements = (): ReactElement[] => {
+    const choosePostageBatch = (postageBatchId: string) => {
+      dispatchGlobalState({ type: 'GLOBAL_POSTAGE_BATCH_SAVE', newValue: postageBatchId })
+    }
+
+    return fetchedPostageBatches.map(postageBatch => {
+      const selected = globalState.postageBatchId === postageBatch.batchID
+      const selectButton = (
+        <a onClick={() => choosePostageBatch(postageBatch.batchID)} className="a-click">
+          select
+        </a>
+      )
+
+      return (
+        <tr key={postageBatch.batchID + selected} className={selected ? 'bold' : ''}>
+          <td>{postageBatch.batchID}</td>
+          <td>{postageBatch.utilization}</td>
+          <td>{selected ? 'selected' : selectButton}</td>
+        </tr>
+      )
+    })
+  }
```

Comment:
I'd prefer this to also be a separate component, but could be done at later stages.